usl_models/usl_models/flood_ml/model.py

```diff
@@ -118,12 +173,82 @@ def forward(
 
         # Concatenate and feed into remaining ConvLSTM and TransposeConv layers
         # [B, n, H', W', k'] -> [B, H, W, 1]
-        lstm_input = self.concat([st_cnn_output, geo_cnn_output, temp_input])
+        lstm_input = tf.concat([st_cnn_output, geo_cnn_output, temp_input], axis=-1)
         lstm_output = self.conv_lstm(lstm_input)
         output = self.output_cnn(lstm_output)
 
         return output
 
-    def call(self):
-        """Runs the entire autoregressive model."""
-        raise NotImplementedError
+    def call(self, inputs):
+        """Runs the entire autoregressive model.
+
+        Args:
+            inputs: A dictionary of input tensors.
+                - "geospatial" (required): [H, W, f]
+                - "temporal" (required): [T]
+                - "spatiotemporal" (optional): a single flood map [H, W]
+
+        Returns:
+            A tuple (all_flood_predictions, max_flood_map) of the flood predictions and
```

Comment:
Should add the `->` return type annotations for this.

Reply:
Done.